src/main/java/core/DataParser.java

```diff
@@ -0,0 +1,98 @@
+package core;
+
+import com.google.gson.*;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+import java.lang.reflect.Type;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Parses between task and string of standard data format.
+ */
+public class DataParser {
```

Comment:
Maybe it would be easier to write your own parser (save as a specific format and parse the saved format into a task list) instead of using existing libraries, since the data to be saved is simple (the tasks in the list)

Reply:
Yes I agree but still amazing job at using the libraries it is really impressive. Maybe just a little overkill for this project.